Kickstarter-iOS/Features/PledgePaymentMethods/Controller/PledgePaymentMethodsViewController.swift

```diff
@@ -180,50 +173,43 @@ final class PledgePaymentMethodsViewController: UIViewController {
       }
     }
 
-    switch data.paymentSheetType {
-    case .setupIntent:
-      PaymentSheet.FlowController.create(
-        setupIntentClientSecret: data.clientSecret,
-        configuration: data.configuration,
-        completion: completion
-      )
-    case .paymentIntent:
-      PaymentSheet.FlowController
-        .create(
-          paymentIntentClientSecret: data.clientSecret,
-          configuration: data.configuration,
-          completion: completion
-        )
-    }
+    PaymentSheet.FlowController.create(
+      setupIntentClientSecret: data.clientSecret,
+      configuration: data.configuration,
+      completion: completion
+    )
   }
 
   private func confirmPaymentResult(with clientSecret: String) {
     guard self.paymentSheetFlowController?.paymentOption != nil else {
       self.viewModel.inputs.shouldCancelPaymentSheetAppearance(state: true)
-
       return
     }
 
     self.paymentSheetFlowController?.confirm(from: self) { [weak self] paymentResult in
 
       guard let strongSelf = self else { return }
 
-      strongSelf.viewModel.inputs.shouldCancelPaymentSheetAppearance(state: true)
-
-      guard let existingPaymentOption = strongSelf.paymentSheetFlowController?.paymentOption else { return }
-
       switch paymentResult {
       case .completed:
-        let paymentDisplayData = PaymentSheetPaymentOptionsDisplayData(
-          image: existingPaymentOption.image,
-          label: existingPaymentOption.label
-        )
-        strongSelf.viewModel.inputs
-          .paymentSheetDidAdd(newCard: paymentDisplayData, clientSecret: clientSecret)
+        // Fetch the stripe payment method ID
```

Comment:
This is a somewhat annoying workaround. When the view model calls `addPaymentSheetPaymentSource` to store the credit card, it does not return a `stripeCardId` in the `UserCreditCard.CreditCard` in the payload. Payments has opened a ticket about this - but for now, we fetch the `stripeCardId` directly from the Stripe API, and pass it back up to the view model.